aten/src/ATen/native/mkldnn/xpu/detail/Matmul.cpp

```diff
@@ -28,6 +29,9 @@ sycl::event matmul(
       dims == mat1.dim() && dims == mat2.dim(),
       "oneDNN input matrixes must have the same ranks");
   TORCH_CHECK(result.defined(), "oneDNN matmul result should be defined");
+  TORCH_CHECK(
+      mat1.scalar_type() != at::kShort,
+      "Short data type matmul is not supported in oneDNN");
```

Comment:
Why does the logic only check `kShort` here? Do you mean the scalar type of `mat1` could be `kShort`?

Reply:
The check added here is we found a ut failure. But it is not relevant to this PR, I will append another PR to add the check and ut fixing. Thanks for reminding.